pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -107,3 +107,11 @@ def _has_already_handler(self, signum: Signals) -> bool:
             return isinstance(signal.getsignal(signum), FunctionType)
         except AttributeError:
             return False
+
+    @staticmethod
+    def _register_signal(signum: Signals, handlers: HandlersCompose) -> None:
+        try:
+            signal.signal(signum, handlers)
+        except ValueError as e:
+            if "signal only works in main thread" not in str(e):
+                raise e
```

Comment:
Could we use https://docs.python.org/3.4/library/threading.html#threading.main_thread here?

Reply:
Done !